src/ReactiveUI/Cocoa/CommonReactiveSource.cs

```diff
@@ -330,7 +330,19 @@ private void SubscribeToSectionInfoChanges(int sectionInfoId, IReadOnlyList<TSec
                                             });
                                     }
 
-                                    this.pendingChanges.Add(Tuple.Create(y.Section, new PendingChange(y.Change)));
```

Comment:
hehe yea I wrote and re-wrote that ticket probably 5 times as my understanding of what was happening kept deepening so the language might not be quite as accurate.  

The same RemoveRange code for me works fine on Android. On Android I'm using a "ReactiveListAdapter" which does really nothing that exciting compared to  CommonReactiveSource.  I find that iOS always requires the more "exciting" solutions :-/ 

The issue is very specific to how iOS does the batch updates and the magic involved with making sure list changes propagate correctly to an iOS batch update

This comment summarizes it all very well 
https://github.com/reactiveui/ReactiveUI/blob/b418df64918da0c78a378af1b846afbab0f4e5ae/src/ReactiveUI/Cocoa/IndexNormalizer.cs#L68

As far as where the logic should live I think you're right. I followed the code further down the rabbit hole and now have the fix summed up with a very small change.

The way I did it at first mainly spawned from  wanting to affect as little as possible. Making changes further down into the core made me nervous that there were other permutations that I'm not quite aware of that I would be breaking but being cautious probably had the opposite result.  For example the if else I did I'm pretty sure was ultimately pointless and could have been summed up with a single line but I was just being cautious :-)

Ultimately the new solution I've checked in seems to be address the problem more at the core.